Parser.fs

```diff
@@ -413,17 +421,17 @@ do parseViewSignatureRef := parseViewLike parseBasicViewSignature ViewSignature.
  * View prototypes.
  *)
 
-/// Parses the LHS of a view prototype.
-let parseViewProtoLhs =
-    pstring "view" >>. ws >>. parseFunc parseTypedTypedVar
 
 /// Parses a view prototype (a LHS followed optionally by an iterator).
 let parseViewProto =
-    parseViewProtoLhs
-    >>= fun lhs ->
-            (wsSemi >>% NoIterator (lhs, false)) <|>
-            (parseIteratorDef .>> wsSemi |>> curry WithIterator lhs)
-    .>> ws
+    // TODO (CaptainHayashi): so much backtracking...
```

Comment:
I think this is the bit that changed `view x(int y, int z)[n]` to `view iter[n] x(int y, int z)`.  The choice used to be after parsing `x(int y, int z)`, but now has to be right after parsing `view`.